docs/architecture/adr-010-posts-module.md

```diff
@@ -0,0 +1,345 @@
+# ADR 010: Posts module
+
+## Changelog
+
+- April 05th, 2022: Initial draft;
+
+## Status
+
+DRAFT
+
+## Abstract
+
+This ADR contains the definition of the `x/posts` module which will allow users to post text contents inside Desmos  subspaces.
+
+## Context
+
+As Desmos is thought to be a protocol to build decentralized socially-enabled applications (or social networks), one of the main features that we MUST make sure exists is the ability for users to create content inside such social networks. When designing this feature, we SHOULD consider the following:
+
+1. a post will always be submitted inside a _subspace_;
+2. a post should allow to specify the minimum amount of information so that any DApp can make use of them;
+3. it is responsibility of the DApp to make sure that the various fields are filled properly upon uploading the post.
+
+## Decision
+
+We will implement a new module named `x/posts` that allows users to perform the following operations inside subspaces that allow them to:
+
+- create a new post
+- edit an existing post
+- participate inside a discussion with a comment or reply to a post
+
+### Types
+Each post MUST always have an _author_ which identifies the user that has created the content. It also MUST always reference a _subspace_ inside which it is created. In order to be valid, a post MUST either contain a _text_ or at least one _attachment_.
+
+Optionally, a post can also have a series of _entities_ that have been parsed out of the text. These allow to identify particular content that should be displayed in custom ways (hashtags, mentions, urls, etc).
+
+#### Post
+
+```protobuf
+// Post contains all the information about a single post
+message Post {
+  // Id of the subspace inside which the post has been created 
+  required uint64 subspace_id = 1;
+
+  // Unique id of the post
+  required uint64 id = 2;
+
+  // Text of the post
+  optional string text = 3;
+
+  // Author of the post
+  required string author = 4;
+
+  // Id of the original post of the conversation
+  optional uint64 conversation_id = 5 [default = 0];
+
+  // Reply settings of this post
+  required ReplySetting reply_settings = 6;
+
+  // Creation date of the post
+  required google.protobuf.Timestamp creation_date = 7;
+
+  // Entities connected to this post
+  optional Entities entities = 8;
+}
+
+// Contains the details of entities parsed out of the post text
+message Entities {
+  repeated Tag hashtags = 1;
+  repeated Tag mentions = 2;
+  repeated Url urls = 3;
+}
+
+// ReplySetting contains the possible reply settings that a post can have
+enum ReplySetting {
+  EVERYONE = 1;
+  FOLLOWERS = 2;
+  MENTIONS = 3;
```

Comment:
I would add also MUTUAL/FRIENDS (mutual following between the creator and the follower).

Reply:
Added